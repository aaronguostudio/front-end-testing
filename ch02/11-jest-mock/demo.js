import axios from 'axios'

export const runCallback = (callback) => {
  callback('abc')
}

export const createObject = (classItem) => {
  new classItem();
}

export const getData = () => {
  return axios.get('fake').then(res => res.data)
}
