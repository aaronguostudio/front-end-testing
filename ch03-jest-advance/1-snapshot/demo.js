export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8083,
    time: new Date()
  }
}

export const generateAnotherConfig = () => {
  return {
    server: 'http://localhost/api/new',
    port: 8084,
    time: new Date()
  }
}
