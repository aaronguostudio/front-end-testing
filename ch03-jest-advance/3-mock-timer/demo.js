export default (fn) => [
  setTimeout(() => {
    fn()
    setTimeout(() => {
      fn()
    }, 30000)
  }, 30000)
]
