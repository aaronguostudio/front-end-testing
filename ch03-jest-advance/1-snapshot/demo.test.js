import { generateConfig, generateAnotherConfig } from './demo'

test('test generateConfig', () => {
  // 仅仅运行一次，生成之后就不再执行了
  // 所以修改了配置文件的话，就需要重新生成快照
  // press u 之后可以更新快照
  // time 这个可以是任何 Date
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date) // any(Number)...
  })
})

test('test generateAnotherConfig', () => {
  expect(generateAnotherConfig()).toMatchSnapshot({
    time: expect.any(Date)
  })
})
