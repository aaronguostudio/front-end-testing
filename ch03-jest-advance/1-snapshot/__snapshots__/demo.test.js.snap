// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generateAnotherConfig 1`] = `
Object {
  "port": 8084,
  "server": "http://localhost/api/new",
  "time": Any<Date>,
}
`;

exports[`test generateConfig 1`] = `
Object {
  "port": 8083,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
